src/context/Auth/util.js

```diff
@@ -0,0 +1,66 @@
+import axios from "axios";
+import { ToastMsg } from "../../components";
+
+export const LoginHandler = async (
+  loginForm,
+  authDispatch,
+  navigate,
+  location
+) => {
+  try {
+    const res = await axios.post("/api/auth/login", loginForm);
+    if (res.status === 200 || res.status === 201) {
+      localStorage.setItem("UserToken", res.data.encodedToken);
+      authDispatch({
+        type: "LOGIN",
+        payload: {
+          user: res.data.foundUser,
+          token: res.data.encodedToken,
+        },
+      });
+      ToastMsg("Log in successful", "success");
+      navigate(location.state?.from?.pathname || "/");
+    }
+  } catch (error) {
+    if (error.response.status == 401) {
+      ToastMsg("Invalid email or password", "error");
+    } else {
+      ToastMsg("Something went wrong,Try again later!", "error");
+    }
+  }
+};
+
+export const handleSingup = async (signupDetails, authDispatch, navigate) => {
+  try {
+    const res = await axios.post("/api/auth/signup", signupDetails);
+    if (res.status === 201) {
+      authDispatch({
+        type: "SIGNUP",
+        payload: {
+          user: res.data.foundUser,
+          token: res.data.encodedToken,
+        },
+      });
+      ToastMsg("Successfully Signed Up", "success");
+      navigate("../login");
```

Comment:
i think here you need to navigate to "/login" after successfully getting the token.

Reply:
it is redirecting to log in still, I will look into it.